plugin/adsk/plugin/plugin.cpp

```diff
@@ -243,6 +244,11 @@ MStatus initializePlugin(MObject obj)
         "mayaUsd_pluginBatchLoad",
         "mayaUsd_pluginBatchUnload");
 
+    // Register to file path editor
+    status = MGlobal::executeCommand("filePathEditor -registerType \"mayaUsdProxyShape.filePath\" "
+                                     "-typeLabel \"mayaUsdProxyShape.filePath\" -temporary");
+    CHECK_MSTATUS(status);
```

Comment:
Was there a reason this wasn't done in the [mayaUsd_pluginUICreation.mel](https://github.com/Autodesk/maya-usd/blob/dev/plugin/adsk/scripts/mayaUsd_pluginUICreation.mel)

And then the also using `deregisterType` in the mayaUsd_pluginUIDeletion (when our plugin is unloaded).

Reply:
I see that all the registrations happening here (registerShape etc.) so I assume this is the place we register stuff? 
Yes I will add  `deregisterType`

Reply:
All the registration stuff here is done using the Maya C++ API. What you are calling is a MEL command. My thinking is that registering the filepatheditor type is a UI bit and thus belongs in our plugin UI creation file?

Reply:
My thinking is that the menu item resulting from this Maya API is similar to the drop down item resulting from Maya import / export api. This is a data-driven api that registers a type pertaining to file path editor, similar to how we register translator types to Maya import / export. 

In other words, we're not creating any UI.

Reply:
Okay far point. We still need the deregister though

Reply:
Hi @seando-adsk , I just realized we don't actually need to deregister it, because I added the `-temporary` flag. With `-temporary` flag the type is only register for the current session. Correct me if I'm wrong

Reply:
Yes that `-temporary` flag means that it will only be for the current session. But if the user unloads the mayausd plugin then we should deregister the filepath.